src/towncrier/test/test_project.py

```diff
@@ -48,25 +41,61 @@ def test_tuple(self):
         version = get_version(temp, "mytestproja")
         self.assertEqual(version, "1.3.12")
 
-    @skipIf(metadata_version is None, "Needs importlib.metadata.")
     def test_incremental(self):
         """
-        An incremental Version __version__  is picked up.
+        An incremental-like Version __version__  is picked up.
+        """
+        temp = self.mktemp()
+        os.makedirs(temp)
+        os.makedirs(os.path.join(temp, "mytestprojinc"))
+
+        with open(os.path.join(temp, "mytestprojinc", "__init__.py"), "w") as f:
+            f.write(
+                """
+class Version:
+    def __init__(self, *version_parts):
+        self.version = version_parts
+
+    def base(self):
+        return '.'.join(map(str, self.version))
+
+
+__version__ = Version(1, 3, 12, "rc1")
+"""
+            )
+        version = get_version(temp, "mytestprojinc")
+        self.assertEqual(version, "1.3.12rc1")
+
+    def test_not_incremental(self):
+        """
+        An incremental-like Version __version__  is picked up, base takes unexpected
+        arguments.
         """
-        pkg = "../src"
+        temp = self.mktemp()
+        os.makedirs(os.path.join(temp, "mytestprojnotinc"))
 
-        with self.assertWarnsRegex(
-            DeprecationWarning, "Accessing towncrier.__version__ is deprecated.*"
-        ):
-            version = get_version(pkg, "towncrier")
+        with open(os.path.join(temp, "mytestprojnotinc", "__init__.py"), "w") as f:
+            f.write(
+                """
+class WeirdVersion:
+    def base(self, some_arg):
+        return "shouldn't get here"
 
-        with self.assertWarnsRegex(
-            DeprecationWarning, "Accessing towncrier.__version__ is deprecated.*"
-        ):
-            name = get_project_name(pkg, "towncrier")
 
-        self.assertEqual(metadata_version("towncrier"), version)
-        self.assertEqual("towncrier", name)
+__version__ = WeirdVersion()
+"""
+            )
+        with self.assertRaises(Exception) as e:
```

Comment:
Which exception does this raise?

Reply:
Literally [`Exception`](https://github.com/twisted/towncrier/pull/491/files#diff-ea234c597bb51dfad3d154ebdeb24af6c75e95ac9a922267f99e7704f1fbde42R72) -- the definition of that exception is existing code, but we can of course tighten this up if it makes sense now.

Reply:
I think that it's ok to keep Exception for now.

Later we can look at raising ClickException or something like this for better CLI integration... but in a separate PR.